Spigot-Server-Patches/0565-Add-install-mvn-option.patch

```diff
@@ -0,0 +1,88 @@
+From: Mariell Hoversholm <proximyst@proximyst.com>
+Date: Sat, 22 Aug 2020 20:53:25 +0200
+Subject: [PATCH] Add -install-mvn option
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
+index bac292e6d8e6cdb239d8accd21eaa25022f5640f..149bdb9444645b94d22d1ea9e751ea6f8eed80a8 100644
+--- a/src/main/java/org/bukkit/craftbukkit/Main.java
++++ b/src/main/java/org/bukkit/craftbukkit/Main.java
+@@ -2,11 +2,18 @@ package org.bukkit.craftbukkit;
+ 
+ import java.io.File;
+ import java.io.IOException;
++import java.io.InputStream; // Paper
++import java.net.URISyntaxException; // Paper
++import java.nio.file.Files; // Paper
++import java.nio.file.Path; // Paper
++import java.nio.file.Paths; // Paper
++import java.nio.file.StandardCopyOption; // Paper
+ import java.text.SimpleDateFormat;
+ import java.util.Arrays;
+ import java.util.Calendar;
+ import java.util.Date;
+ import java.util.List;
++import java.util.Properties; // Paper
+ import java.util.concurrent.TimeUnit;
+ import java.util.logging.Level;
+ import java.util.logging.Logger;
+@@ -145,6 +152,8 @@ public class Main {
+                         .ofType(String.class)
+                         .defaultsTo("Unknown Server")
+                         .describedAs("Name");
++
++                acceptsAll(asList("install-mvn"), "Install the jar to maven local");
+                 // Paper end
+             }
+         };
+@@ -196,6 +205,49 @@ public class Main {
+             }
+         } else if (options.has("v")) {
+             System.out.println(CraftServer.class.getPackage().getImplementationVersion());
++            // Paper start - install to mvn local
++        } else if (options.has("install-mvn")) {
++            Properties pomProps = new Properties();
++            // TODO: Replace with io.papermc.paper when the time is right
++            try(InputStream stream = Main.class.getResourceAsStream("/META-INF/maven/com.destroystokyo.paper/paper/pom.properties")) {
++                pomProps.load(stream);
++            } catch (IOException ex) {
++                System.err.println("Could not load pom as properties");
++                ex.printStackTrace();
++                return;
++            }
++            String version = pomProps.getProperty("version", Main.class.getPackage().getImplementationVersion());
++
++            Path jarPath;
++            try {
++                jarPath = Paths.get(Main.class.getProtectionDomain().getCodeSource().getLocation().toURI());
++            } catch (URISyntaxException e) {
++                System.err.println("Could not get running jar location");
++                e.printStackTrace();
++                return;
++            }
++
++            // TODO: Replace with `"io", "papermc", "paper"` when the time is right
++            Path mavenRepo = Paths.get(System.getProperty("user.home"), ".m2", "repository", "com", "destroystokyo", "paper", "paper", version).toAbsolutePath();
```

Comment:
This shouldn't be hardcoded due to the whole nature of this being configurable in maven itself. The best option, imho, would be to just copy the pom file into the folder and invoke `mvn install:file` or something